### PR TITLE
Simplify handling of profile handling in test cases

### LINKED ISSUE #27545: Simplify handling of profile features in test cases
### Description

When updating profiles, a lot of files need to change, even if they usually test different things, like CLI options available in the help texts.

### Discussion

_No response_

### Motivation

Simplify contributions, and minimize conflicts between PRs.

### Details

I'll provide a PR for this

### PR DESCRIPTION
**Pull request by ahus1**
Closes #27545

### CODE REVIEW

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java [hunk 1]
```diff
@@ -172,7 +172,9 @@ private void assertSingleJvmStarted(CLIResult run) {
 
     private void assertHelp(CLIResult result) {
         try (NamedEnvironment env = KcNamerFactory.asWindowsOsSpecificTest()) {
-            Approvals.verify(result.getOutput());
+            // normalize the output to prevent changes around the feature toggles to mark the output to differ
+            String output = result.getOutput().replaceAll("((Disables|Enables) a set of one or more features. Possible values are: )[^.]*", "$1<...>");
```
Comment (vmuzikar) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java:176]: IMHO we should check there are at least some features printed. Something like this perhaps?
```
([\S-]+\[:v\d], )+[\S-]+\[:v\d]\.
```

**Review by vmuzikar — COMMENTED** [review]
@ahus1 Thank you, I like this idea!

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java (hunk 1 above)
Comment (ahus1) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java]: I tried to get the regex right, but it turned out to be quite complicated. I found that that it can word-wrap the options with the brackets just before the opening bracket. 

So my alternative would be to check that there are at least 30 characters of options. See current commit for a regex: 

```
((Disables|Enables) a set of one or more features. Possible values are: )[^.]{30,}
```

File: quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.windows.approved.txt [hunk 2]
```diff
@@ -165,6 +149,11 @@ HTTP(S):
 --http-max-queued-requests <requests>
                      Maximum number of queued HTTP requests. Use this to shed load in an overload
                        situation. Excess requests will return a "503 Server not Available" response.
+--http-pool-max-threads <threads>
```
Comment (mabartos) [inline quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.windows.approved.txt:152]: This seems unrelated to this PR, right?

**Review by mabartos — COMMENTED** [review]
@ahus1 Looks good! One thing - are we confident that the testing of feature states has proper test coverage? As for the approval tests, we were quite confident the features are disabled as expected.

File: quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.windows.approved.txt (hunk 2 above)
Comment (Pepo48) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.windows.approved.txt:152]: @mabartos that was added as a part of the https://github.com/keycloak/keycloak/pull/26659 PR.

**Comment by Pepo48** [comment]
LGTM as well. Thanks, @ahus1.

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java (hunk 1 above)
Comment (vmuzikar) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java]: Ok, let's not overcomplicate things. That compromise works for me. Thanks!

**Review by vmuzikar — APPROVED** [review]

**Review by mabartos — APPROVED** [review]
After going through the Profile tests, I'd say the test coverage is sufficient. Even when the test coverage slightly decreased, the maintainability, on the other hand, rapidly increased, so +1000.

@ahus1 Nice work!

**Comment by ahus1** [comment]
Thank you for agreeing on this pragmatic approach!

@jonkoops - as we have been discussion this part in the past, I hope it makes your life easier as well.

**Comment by jonkoops** [comment]
You guys are livesavers! Thanks all!
